src/transformers/generation_beam_search.py

```diff
@@ -330,7 +367,430 @@ def finalize(
                 best_scores[i * self.num_beam_hyps_to_keep + j] = best_score
 
         # prepare for adding eos
-        sent_max_len = min(sent_lengths.max().item() + 1, max_length)
+        sent_lengths_max = sent_lengths.max().item() + 1
+        sent_max_len = min(sent_lengths_max, max_length) if max_length is not None else sent_lengths_max
+        decoded: torch.LongTensor = input_ids.new(batch_size * self.num_beam_hyps_to_keep, sent_max_len)
+        # shorter batches are padded if needed
+        if sent_lengths.min().item() != sent_lengths.max().item():
+            assert pad_token_id is not None, "`pad_token_id` has to be defined"
```

Comment:
Could we please transform this into:
```suggestion
            if pad_token_id is None:
                raise ValueError("`pad_token_id` has to be defined")
```

Reply:
This code is very sensitive and should not be changed at all. Is GitHub just showing a very bad diff or was some code changed here?